saleor/graphql/order/tests/queries/test_order_with_filter.py

```diff
@@ -1249,3 +1249,79 @@ def test_order_query_with_filter_not_allow_numbers_and_ids_together(
     # then
     assert content["errors"][0]["message"] == error_message
     assert not content["data"]["orders"]
+
+
+def test_order_query_with_filter_by_checkout_token(
+    orders_query_with_filter,
+    staff_api_client,
+    permission_manage_orders,
+    orders_from_checkout,
+    checkout,
+):
+    # given
+    variables = {
+        "filter": {
+            "checkoutToken": checkout.token,
+        }
+    }
+
+    # when
+    response = staff_api_client.post_graphql(
+        orders_query_with_filter, variables, permissions=(permission_manage_orders,)
+    )
+
+    # then
+    content = get_graphql_content(response)
+    order_data = content["data"]["orders"]["edges"]
+
+    assert content["data"]["orders"]["totalCount"] == 4
+    for order in orders_from_checkout:
+        if order.checkout_token == checkout.token:
```

Comment:
This `if` can cause false positive. If you don't get any order with checkout_token, the test will pass.
Also personally I am not a fan of having `if` condition inside the test. For me it means, that the test have more than single way to finish.

Reply:
IMO, you could do something like this:
```
content = get_graphql_content(response)
order_data = content["data"]["orders"]["edges"]
orders_ids = set([edge["id"] for edge in order_data])
assert len(orders_ids.intersection(set([graphene.Node.to_global_id("Order", order.id) for order in orders_from_checkout]))) == 4
```
(needs to be refactored but I think that you could use `set` here)